### PR TITLE
Implement completionItem/resolve, textDocument/codeAction, textDocument/codeLens and codeLens/resolve

### LINKED ISSUE #9: Add support for completionItem/resolve request on server
The `LanguageServer` trait is currently missing support for the [`completionItem/resolve`](https://microsoft.github.io/language-server-protocol/specification#completionItem_resolve) request which is often used in conjunction with the `textDocument/completion` request (see #10).

### PR DESCRIPTION
**Pull request by icsaszar**
Resolves #9

### CODE REVIEW

**Comment by icsaszar** [comment]
@ebkalderon I'm also working on `textDocument/codeAction`, `textDocument/codeLens` and `codeLens/resolve`. Is it ok if I push them here to merge all of these at once?

File: src/delegate.rs [hunk 1]
```diff
@@ -87,6 +87,9 @@ pub trait LanguageServerCore {
     #[rpc(name = "textDocument/completion", raw_params)]
     fn completion(&self, params: Params) -> BoxFuture<Option<CompletionResponse>>;
 
+    #[rpc(name = "completionItem/resolve", raw_params)]
+    fn resolve(&self, params: Params) -> BoxFuture<CompletionItem>;
```
Comment (icsaszar) [inline src/delegate.rs]: I just realized this is a bad idea, because it clashes with `code_lens_resolve`, I'll change it to `completion_resolve`.

Comment (ebkalderon) [reply src/delegate.rs]: No worries! I was just about to suggest the same thing. Thanks for preempting me in that regard, haha.

**Comment by ebkalderon** [comment]
Also, this repository is rebasing merges and not squashing them, meaning we still get to preserve commit-by-commit history. So it's not like having them bundled in one PR ruins the readability of history in `master` for us. 😄 

Ideally, we should have each newly added method be in a separate commit. Feel free to ping me again for a second review, if needed! Thanks for opening this pull request, @icsaszar.

**Comment by icsaszar** [comment]
@ebkalderon I've added the new methods, everything should be complete now. 

This also brings up a small new issue: not that methods are optional by default, it is easy to forget to implement them for Box in lib.rs.

If you're ok with it, I can try to write a macro (I don't have too much experience with macros either) to automatically fill out the impls for Box.

**Comment by ebkalderon** [comment]
I agree that we should write a macro, but I doubt `auto_impl` will work with `async-trait` since the raw syntax generated by the procedural macro will likely not match up with what `auto_impl` expects. But I believe that should land in a separate PR from this one, since it's not directly related to the task at hand.

Otherwise, thanks a lot for the helpful PR! I think it's ready to merge. 👍

**Comment by ebkalderon** [comment]
Resolves #11 and affects #10.

**Pull request merged**